8310847: [Mac] Silence OpenGL deprecation warnings

beldenfox opened the pull request:
The Mac build issues a lot of warnings including several related to the deprecation of OpenGL in macOS 10.14. Now that the deployment target for JavaFX is 11.0 across the board these warnings are showing up on both M1 and Intel builds. Luckily Apple provides a simply way to silence these warnings by setting a single C preprocessor define.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8310847](https://bugs.openjdk.org/browse/JDK-8310847): [Mac] Silence OpenGL deprecation warnings (**Bug** - P4)


### Reviewers
 * [Ambarish Rapte](https://openjdk.org/census#arapte) (@arapte - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1161/head:pull/1161` \
`$ git checkout pull/1161`

Update a local copy of the PR: \
`$ git checkout pull/1161` \
`$ git pull https://git.openjdk.org/jfx.git pull/1161/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1161`

View PR using the GUI difftool: \
`$ git pr show -t 1161`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1161.diff">https://git.openjdk.org/jfx/pull/1161.diff</a>

</details>


### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1161#issuecomment-1605670595)

arapte commented:
All the OpenGL warnings get silenced except following two:

1. modules/javafx.graphics/src/main/native-prism-es2/macosx/MacOSXWindowSystemInterface.m:120:24: warning: 'setView:' is deprecated: first deprecated in macOS 10.14 - Use NSOpenGLView to provide OpenGL content in a Cocoa app. [-Wdeprecated-declarations]

2. modules/javafx.graphics/src/main/native-prism-es2/macosx/MacOSXWindowSystemInterface.m:181:57: warning: 'NSOpenGLCPSwapInterval' is deprecated: first deprecated in macOS 10.14 [-Wdeprecated-declarations]

Could you please take a look why are these not silenced.
Otherwise looks good.

beldenfox commented:
@arapte NSOpenGLCPSwapInterval is deprecated in favor of NSOpenGLContextParameterSwapInterval. I've updated the PR. Apparently Apple deprecated OpenGL and cleaned up the API all at the same time.

Prism creates an NSOpenGLContext and then attaches it to the NSView using `setView:`. This approach is deprecated, the recommended way is to create an NSOpenGLView or CAOpenGLLayer and use the associated context. I don't understand Prism at all so can't really address this. (In fact I'm confused. In Glass the 3D view is an NSOpenGLView and also creates a CAOpenGLLayer which seems redundant. Then there's this Prism code that is creating an NSOpenGLContext and attaching it to the NSView? I'm sure it all makes sense but it would nice to see some documentation on how this all hangs together.)

arapte reviewed (approved):
Change looks good.

beldenfox commented:
/integrate

kevinrushforth commented:
/sponsor
